overlord/snapstate/handlers.go

```diff
@@ -1942,6 +1942,128 @@ func (m *SnapManager) doPreferAliases(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// nearlyReady returns whether all lanes (other than me) are Ready.
+func nearlyReady(me string, change *state.Change) bool {
+	for _, task := range change.Tasks() {
+		if me == task.ID() {
+			// ignore self
+			continue
+		}
+		if !task.Status().Ready() {
+			return false
+		}
+	}
+	return true
+}
+
+// laneSnaps returns the instance names from the first SnapSetup in each
+// non-zero Done lane in the refresh batch leading up the given task
+func laneSnaps(reTask *state.Task) []string {
+	tid := reTask.ID()
+	laneSnaps := map[int]string{}
+	// change.Tasks() preserves the order tasks were added, otherwise it all falls apart
+	for _, task := range reTask.Change().Tasks() {
+		if task.ID() == tid {
+			// we've reached ourselves; we don't care about anything beyond this
+			break
+		}
+		if task.Kind() == "check-rerefresh" {
+			// we've reached a previous check-rerefresh (but not ourselves).
+			// Only snaps in tasks after this point are of interest.
+			laneSnaps = map[int]string{}
+		}
+		lanes := task.Lanes()
+		if len(lanes) != 1 || lanes[0] == 0 {
+			// can't happen, really
```

Comment:
This does happen in tests (reRefreshSuite.TestLaneSnapsRerefreshResets) it seems. Should something be updated? If this ever happens (our task code seems to allow that), should we do something? warn? panic? Or will things just continue working fine?

Reply:
Ah, I should've edited the comment: `len(lanes) != 1` can't happen; `lanes[0] == 0` happens all the time.